Dockerfile

```diff
@@ -41,9 +49,15 @@ COPY --from=pg-build /home/nonroot/pg_install/v16/include/postgresql/server pg_i
 COPY --from=pg-build /home/nonroot/pg_install/v17/include/postgresql/server pg_install/v17/include/postgresql/server
 COPY --from=pg-build /home/nonroot/pg_install/v16/lib                       pg_install/v16/lib
 COPY --from=pg-build /home/nonroot/pg_install/v17/lib                       pg_install/v17/lib
+COPY --from=plan     /home/nonroot/recipe.json                              recipe.json
+
+ARG ADDITIONAL_RUSTFLAGS=""
+
+RUN set -e \
+    && RUSTFLAGS="-Clinker=clang -Clink-arg=-fuse-ld=mold -Clink-arg=-Wl,--no-rosegment -Cforce-frame-pointers=yes ${ADDITIONAL_RUSTFLAGS}" cargo chef cook --release --recipe-path recipe.json
```

Comment:
We need `--locked` here too, right? Otherwise, chef will cook crates that are not used.

Reply:
`--locked` has different effects for `cargo install` on one hand and `cargo build` / `cargo chef cook` on the other hand. In the context of `cargo build`, it means to fail if `Cargo.lock` isn't up to date with `Cargo.toml`, only in `cargo install` it is required to actually use `Cargo.lock`. Not using `--locked` would mean `cargo chef cook` would potentially cook crates that aren't going to be used, but then `cargo build` would fail because `Cargo.lock` isn't up to date. If `Cargo.lock` is up to date, then it doesn't make a difference in the result.

While I don't agree that we *need* it, it's better to fail fast if  `Cargo.lock` is out of date, so I'll add it. Good catch!